File: solutions/elixir/darts/1/lib/darts.ex
defmodule Darts do
  @type position :: {number, number}

  @doc """
  Calculate the score of a single dart hitting a target
  """
  @spec score(position) :: integer
  def score({x, y}) do
    dist = (x ** 2 + y ** 2) ** 0.5
    cond do
      dist > 10 -> 0
      dist > 5 -> 1
      dist > 1 -> 5
      true -> 10
    end
  end
end

File: solutions/ocaml/darts/1/darts.ml
let score (x: float) (y: float): int =
  let dist = 
    (Float.pow x 2.0)
    |> Float.add (Float.pow y 2.0)
    |> Float.sqrt
  in
  match () with
  | _ when dist > 10.0 -> 0
  | _ when dist > 5.0 -> 1
  | _ when dist > 1.0 -> 5
  | _ -> 10